experimental/dds/tree2/src/test/snapshots/utils.ts

```diff
@@ -4,20 +4,134 @@
  */
 
 import { strict as assert } from "assert";
-import { promises as fs, existsSync } from "fs";
-import { Serializable } from "@fluidframework/datastore-definitions";
+import { promises as fs, existsSync, rmSync, mkdirSync } from "fs";
+import { ISummaryTree, SummaryType, SummaryObject } from "@fluidframework/protocol-definitions";
+import { Uint8ArrayToString } from "@fluid-internal/client-utils";
 
-const numberOfSpaces = 4;
+const numberOfSpaces = 2;
 
-export async function createSnapshot(path: string, data: Serializable): Promise<void> {
-	const dataStr = JSON.stringify(data, undefined, numberOfSpaces);
+function getSummaryTypeName(summaryObject: SummaryObject): "blob" | "tree" {
+	const type =
+		summaryObject.type === SummaryType.Handle ? summaryObject.handleType : summaryObject.type;
+
+	switch (type) {
+		case SummaryType.Blob:
+		case SummaryType.Attachment:
+			return "blob";
+		case SummaryType.Tree:
+			return "tree";
+		default:
+			throw new Error(`Unknown type: ${type}`);
+	}
+}
+
+function serializeTree(parentHandle: string, tree: ISummaryTree, rootNodeName: string) {
```

Comment:
```suggestion
/**
 * Rather than JSON.stringify the entire `ISummaryTree` for the purpose of writing it to disk,
 * this function more closely resembles how the summary gets serialized to storage.
 * This helps avoid confusion on problems such as double-JSON-stringification.
 */
function serializeTree(parentHandle: string, tree: ISummaryTree, rootNodeName: string) {
```